unit-tests/syncer/test-ts-eof.py

```diff
@@ -0,0 +1,150 @@
+# License: Apache 2.0. See LICENSE file in root directory.
+# Copyright(c) 2021 Intel Corporation. All Rights Reserved.
+
+import pyrealsense2 as rs
+from rspy import log, test
+import sw
+
+
+# The timestamp jumps are closely correlated to the FPS passed to the video streams:
+# syncer expects frames to arrive every 1000/FPS milliseconds!
+sw.fps_c = sw.fps_d = 60
+sw.init()
+
+import tempfile, os
+temp_dir = tempfile.TemporaryDirectory( prefix = 'recordings_' )
+filename = os.path.join( temp_dir.name, 'rec.bag' )
+recorder = rs.recorder( filename, sw.device )
+
+sw.start()
+
+#############################################################################################
+#
+test.start( "Init" )
+
+# It can take a few frames for the syncer to actually produce a matched frameset (it doesn't
+# know what to match to in the beginning)
+
+sw.generate_depth_and_color( frame_number = 0, timestamp = 0 )
+sw.expect( depth_frame = 0 )                          # syncer doesn't know about color yet
+sw.expect( color_frame = 0, nothing_else = True )     # less than next expected of D
+#
+# NOTE: if the syncer queue wasn't 100 (see above) then we'd only get the color frame!
```

Comment:
Why is that?
The syncer should output 1 frameset with depth + color no?
Why not `sw.expect( depth_frame = 0,  color_frame = 0, nothing_else = True )  `